### DISCUSSION TITLE
Episode 160: Rust & Safety at Adobe with Sean Parent

### DISCUSSION THREAD

**Discussion by codereport** [codereport/adsp2#52] [Episode]
Episode Link: https://adspthepodcast.com/2023/12/15/Episode-160.html

Feel free to write comments or ask questions 🙂

**Comment by tmr232** [comment]
Notepad.exe _does_ autosave now.

**Reply by codereport** [reply]
Really, i feel like i have lost stuff recently to the notepad void.

**Reply by tmr232** [reply]
Well, I guess it's _almost_ autosave (on Windows 11)
![image](https://github.com/codereport/adsp2/assets/1386239/c34afa1b-5a06-4f58-8af9-fbb589ee2907)

**Comment by FerdinandStapenhorst** [comment]
About the fact that software is built with the need of being able to restart with everything reset - here’s a joke about it:

An engineer, a physicist, and a programmer are riding in a car. Suddenly, the car breaks down. The engineer suggests, "It's probably the fuel injection system, I'll check it out right away." The physicist thinks, "Nah, it's just a loose cable." The programmer then suggests, "Let's all get out, get back in, and it should work again." 🙄
The question is: is it funny or sad?

**Comment by JVApen** [comment]
The focus regarding safety was on memory safety, though it ain't the only source of bugs. If your phone update results in an infinite loop, you won't be able to use it either. If you have type mismatch in semi dynamic language, it also aborts the program. Shouldn't the focus be on removing all sources of bugs rather than just memory safety?
I feel Bjarne described the whole of the problem quite well at cppcon: https://youtu.be/I8UvQKvOSSw

**Reply by tmr232** [reply]
[Memory Safety is a Red Herring](https://steveklabnik.com/writing/memory-safety-is-a-red-herring) is a good take on this, IMO.

> a focus on memory safe languages (MSLs) versus non memory-safe languages is a bit of a red herring. The actual distinction is slightly bigger than that: languages which have defined behavior by default, with a superset where undefined behavior is possible, vs languages which allow for undefined behavior anywhere in your program.
